presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -387,6 +403,9 @@ private Page extractOutput()
     {
         // INVARIANT: pagesIndex contains the full grouped & sorted data for one or more partitions
 
+        // Index is already prepared
+        windowInfo.addIndex(pagesIndex);
```

Comment:
If you output `avg` and `std.dev` in `EXPLAIN ANALYZE`, then individual data points won't be required. You can compute both of those stats online.